lib/utils/queryable.js

```diff
@@ -119,14 +144,33 @@ const setter = ({ data, key, value, force }) => {
   // ['foo', 'bar', 'baz'] -> { foo: { bar: { baz:  {} } }
   const keys = parseKeys(key)
   const setKeys = (_data, _key) => {
-    // handles array indexes, making sure the new array is created if
-    // missing and properly casting the index to a number
-    const maybeIndex = Number(_key)
-    if (!Number.isNaN(maybeIndex)) {
+    // handles array indexes, converting valid integers to numbers,
+    // note that occurences of Symbol(append) will throw,
+    // so we just ignore these for now
+    let maybeIndex = Number.NaN
+    try {
+      maybeIndex = Number(_key)
+    } catch (err) {}
+    if (!Number.isNaN(maybeIndex))
       _key = maybeIndex
-      if (!Object.keys(_data).length)
-        _data = []
-    }
+
+    // creates new array in case key is an index
+    // and the array obj is not yet defined
+    const keyIsAnArrayIndex = _key === maybeIndex || _key === _append
+    const dataHasNoItems = !Object.keys(_data).length
+    if (keyIsAnArrayIndex && dataHasNoItems)
+      _data = []
+
+    // converting from array to an object is also possible, in case the
+    // user is using force mode, we should also convert existing arrays
+    // to an empty object if the current _data is an array
+    if (force && Array.isArray(_data) && _key !== maybeIndex)
```

Comment:
why only in force mode? in non-force mode, is there an error for this code path?

Reply:
yep, it's an error:

```
$ npm pkg set foo[0]=foo foo.bar=bar
npm ERR! code EOVERRIDEVALUE
npm ERR! Property foo.bar already has a value in place.

npm ERR! A complete log of this run can be found in:
npm ERR!     $HOME/.npm/_logs/2021-07-13T20_17_50_539Z-debug.log

$ npm pkg set foo[0]=foo foo.bar=bar --force
npm WARN using --force Recommended protections disabled.

$ git diff
diff --git a/package.json b/package.json
index 38c21cf..fc9db2a 100644
--- a/package.json
+++ b/package.json
@@ -6,5 +6,8 @@
   },
   "dependencies": {
     "@ruyadorno/slow-lifecycle-script": "^1.0.3"
+  },
+  "foo": {
+    "bar": "bar"
   }
 }
```

Reply:
"Property foo.bar already has a value in place." seems like an incorrect error message for this case.

However also, why wouldn't, with force, it result in:
```
$ git diff
diff --git a/package.json b/package.json
index 38c21cf..fc9db2a 100644
--- a/package.json
+++ b/package.json
@@ -6,5 +6,8 @@
   },
   "dependencies": {
     "@ruyadorno/slow-lifecycle-script": "^1.0.3"
+  },
+  "foo": {
+    "0": "foo",
+    "bar": "bar"
   }
 }
```

Reply:
good points! updated ✅ looks like this now:

```
$ npm pkg set foo[0]=foo foo.bar=bar
npm ERR! code ENOADDPROP
npm ERR! Can't add property foo.bar to an Array.

npm ERR! A complete log of this run can be found in:
npm ERR!     $HOME/.npm/_logs/2021-07-13T21_07_18_074Z-debug.log

$ npm pkg set foo[0]=foo foo.bar=bar --force
npm WARN using --force Recommended protections disabled.

$ git diff
diff --git a/package.json b/package.json
index 38c21cf..0f1437f 100644
--- a/package.json
+++ b/package.json
@@ -6,5 +6,9 @@
   },
   "dependencies": {
     "@ruyadorno/slow-lifecycle-script": "^1.0.3"
+  },
+  "foo": {
+    "0": "foo",
+    "bar": "bar"
   }
 }
```